webapp/src/DOMJudgeBundle/Controller/AnalysisController.php

```diff
@@ -0,0 +1,393 @@
+<?php
+
+namespace DOMJudgeBundle\Controller;
+
+use DOMJudgeBundle\Service\DOMJudgeService;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use DOMJudgeBundle\Entity\Problem;
+use DOMJudgeBundle\Entity\Team;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+
+/**
+ * @Route("/jury/analysis")
+ * @Security("has_role('ROLE_JURY')")
+ */
+class AnalysisController extends Controller
+{
+    /**
+     * @var DOMJudgeService
+     */
+    private $DOMJudgeService;
+
+    private static function set_or_increment(array &$array, string $index) {
+      if (!array_key_exists($index, $array)) {
+        $array[$index] = 0;
+      }
+      $array[$index] = $array[$index] +1;
+    }
+
+    public function __construct(DOMJudgeService $DOMJudgeService)
+    {
+        $this->DOMJudgeService = $DOMJudgeService;
+    }
+
+    /**
+     *@Route("/", name="analysis_index")
+     */
+    public function indexAction(Request $request)
+    {
+        $em = $this->getDoctrine()->getManager();
+        $contest = $this->DOMJudgeService->getCurrentContest();
+
+        // First collect information about problems in this contest
+        $problems = $contest->getProblems(); // TODO: this gets ContestProblem, but we need Problem as well
```

Comment:
I was thinking about this actually, as for the API it would also result in N+1 queries. I think for contestproblem it makes sense to eagerly load the problems as well in the entity?